CONTRIBUTING.rst

```diff
@@ -124,15 +124,36 @@ See the relevant sections above on how to do this.
 Setting up the environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Follow the steps in the :ref:`installation_instructions` > "From Source" section
-to set up your environment, install the required development dependencies, and
-run the tests.
+To contribute, you will first have to run through some steps:
+
+- Set up your environment by forking the repository ((`Github doc on
+  forking and
+  cloning <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo>`__)).
+- Create a new virtual environment either with
+  `venv <https://docs.python.org/3/library/venv.html>`__ or
+  `conda <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`__,
+  then activate it.
+- While at the root of your local copy of skrub and within the new env,
+  install the required development dependencies by running
+  ``pip install --editable ".[dev, lint, test, doc]"``.
+- Run all the tests by running ``pytest -s skrub/tests``.
+- Run ``pre-commit install`` to activate some checks that will run every
+  time you do a ``git commit`` (mostly, formatting checks).
+
+Now that the development environment is ready, you may start working on
+the new issue by creating a new branch:
 
-When starting to work on a new issue, it's recommended to create a new branch:
+.. code:: sh
 
-.. code:: console
+   git checkout -b my-branch-name-eg-fix-issue-123
+   # make some changes
+   git add ./the/file-i-changed
+   git commit -m "my message"
+   git push --set-upstream origin my-branch-name-eg-fix-issue-123
 
-   git switch -c branch_name
+At this point, if you visit again the `pull requests
+page <https://github.com/skrub-data/skrub/pulls>`__ github should show a
+banner asking if you want to open a pull request from your new branch.
```

Comment:
also I don't think the "Implementation guidelines" below is very helpful

Reply:
should it be removed?

Reply:
@jeromedockes Do you think it's too generic? Otherwise, I believe it summarizes our first principles quite well for a new contributor.

> Avoid using binary extensions, Cython, or other compiled languages.

This one for example is not obvious and it's good to have it highlighted there.